### ISSUE TITLE
Switch shipping service from grpc to http

### ISSUE THREAD

**Issue by cartersocha** [open-telemetry/opentelemetry-demo#575]
To improve build times we would like to refactor from using grpc in the shipping service to http

**Comment by cartersocha** [comment]
@GaryPWhite

**Comment by GaryPWhite** [comment]
hi :)

**Comment by GaryPWhite** [comment]
will also post timing when removing tonic/grpc

**Comment by GaryPWhite** [comment]
hey folks! I was able to find some... troubling things when testing which package would reduce the time the most.

I investigated by commenting build scripts / our crate code and playing with the installed dependencies. The biggest bottleneck, I noticed, was the `Updating crates.io index` -- it took between 75-90 seconds of the ~180 second compilation as I was testing it out. Even with _one_ package being installed, this was the majority of the time that `cargo build` would spend on compiling and installing.

turns out the compilation of the app takes ~50s out of a ~180 second compilation.

Why this is especially important to this issue: a package we _cannot_ remove, [opentelemetry_oltp](https://crates.io/crates/opentelemetry-otlp/0.11.0/dependencies) (how metrics are sent to the otel collector) depends on grpc anyways. It doesn't really make sense as a _time saving measure_ to switch from grpc to anything else, since we have to install [`tonic`](https://crates.io/crates/tonic/0.8.3)(grpc in rust) and a couple other dependencies that are shared between our grpc implementation and others.

Broad takeaway before I refactor most of the app: this is not a meaningful time save. compiling almost no code between the packages showed almost the exact same time with/without grpc -- I suspect the time save lies in caching our crates.io index. We can chat more about this in the next meeting or on this issue!
